### PR TITLE
[Improvements]: various clang-tidy fixes and typo

### PR DESCRIPTION
**Pull request by Milerius**
## Description

Clang tidy various fixes, typo fixes

Aion/Algorand/Aeternity/Binance/Bitcoin

## How to test

Run C++ unit tests

## Types of changes

Improvements

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: src/Binance/Entry.cpp [hunk 1]
```diff
@@ -24,7 +23,7 @@ std::string Entry::deriveAddress([[maybe_unused]] TWCoinType coin, const PublicK
 Data Entry::addressToData([[maybe_unused]] TWCoinType coin, const std::string& address) const {
     Address addr;
     if (!Address::decode(address, addr)) {
-        return Data();
+        return {};
```
Comment (catenocrypt) [inline src/Binance/Entry.cpp:26]: Can you explain the argument for `{}` instead of `Data()`?

File: src/Bitcoin/InputSelector.h [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 namespace TW::Bitcoin {
 
-template <typename TypeWithAmount> // TypeWithAmount has to have a uint64_t amount
+template <typename TypeWithAmount> // TypeWithAmount has to have an uint64_t amount
```
Comment (catenocrypt) [inline src/Bitcoin/InputSelector.h:17]: Now this is pure lingustics, but:
'a U-Int' but
'an unsigned integer'
I would use 'a' as uint is an abbreviation.
https://oneminuteenglish.org/en/a-or-an-before-u/

**Review by catenocrypt — APPROVED** [review]
Excellent improvements!

File: src/Binance/Entry.cpp (hunk 1 above)
Comment (Milerius) [reply src/Binance/Entry.cpp:26]: Data is already specified in the return type, with our current clang-tidy settings it produce a redundant type warning

File: src/Bitcoin/InputSelector.h (hunk 2 above)
Comment (Milerius) [reply src/Bitcoin/InputSelector.h:17]: will fix, thanks

**Review by miloserdow — APPROVED** [review]

File: src/Binance/Signer.h [hunk 3]
```diff
@@ -25,7 +26,7 @@ class Signer {
     Proto::SigningInput input;
 
     /// Initializes a transaction signer.
-    explicit Signer(const Proto::SigningInput& input) : input(input) {}
+    explicit Signer(Proto::SigningInput  input) : input(std::move(input)) {}
```
Comment (hewigovens) [inline src/Binance/Signer.h:29]:
```suggestion
    explicit Signer(Proto::SigningInput input) : input(std::move(input)) {}
```

File: src/Bitcoin/Script.h [hunk 4]
```diff
@@ -29,8 +30,8 @@ class Script {
     template <typename It>
     Script(It begin, It end) : bytes(begin, end) {}
 
-    /// Initializaes a script with a collection of raw bytes by moving.
-    explicit Script(const Data& bytes) : bytes(bytes) {}
+    /// Initializes a script with a collection of raw bytes by moving.
+    explicit Script(Data  bytes) : bytes(std::move(bytes)) {}
```
Comment (hewigovens) [inline src/Bitcoin/Script.h:34]:
```suggestion
    explicit Script(Data bytes) : bytes(std::move(bytes)) {}
```

File: src/Bitcoin/SignatureBuilder.h [hunk 5]
```diff
@@ -59,13 +60,13 @@ class SignatureBuilder {
     /// Initializes a transaction signer with signing input.
     /// estimationMode: is set, no real signing is performed, only as much as needed to get the almost-exact signed size 
     SignatureBuilder(
-        const SigningInput& input,
-        const TransactionPlan& plan,
+        SigningInput  input,
+        TransactionPlan  plan,
```
Comment (hewigovens) [inline src/Bitcoin/SignatureBuilder.h:64]:
```suggestion
        SigningInput input,
        TransactionPlan plan,
```

**Review by hewigovens — DISMISSED** [review]
Looks good, some extra spacing

**Review by catenocrypt — APPROVED** [review]

File: src/Bitcoin/InputSelector.h (hunk 2 above)
Comment (catenocrypt) [reply src/Bitcoin/InputSelector.h:17]: As I said, it depends on how you read it: 'a you-int-sixtyfor' or 'an unsigned-integer-sixtyfor'. No definite correct answer

**Review by hewigovens — APPROVED** [review]

**Review by catenocrypt — APPROVED** [review]

**Pull request merged**